lib/galaxy/tools/__init__.py

```diff
@@ -2772,10 +2773,8 @@ def json_fix( val ):
         return [ json_fix( v ) for v in val ]
     elif isinstance( val, dict ):
         return dict( [ ( json_fix( k ), json_fix( v ) ) for ( k, v ) in val.iteritems() ] )
-    elif isinstance( val, unicode ):
-        return val.encode( "utf8" )
     else:
-        return val
+        return unicodify(val, "utf8" )
```

Comment:
+1 to remove `json_fix` from here and import if from `galaxy.util.json` .

I think the fix for #1702 should be different, I'll comment there.